BackgammonPlay.py

```diff
@@ -0,0 +1,256 @@
+            self.turn_in_game(player)
+
+    def level(self, player_now):
+        if player_now is p1:
+            if (p1.get_sign() not in self.board[:-6]):
+                return 2  # level 2
+            return 1  # level 1
+        if (p2.get_sign() not in self.board[6:]):
+            return 2
+        return 1
+
+    def play_level_2(self, index, step, player_now):
+        step = step - 1
+        if player_now is p2:
+            if ((index == step and p2.get_sign() in self.board.get_board()[index]) or (
+                    index < step and p2.get_sign() not in self.board.get_board()[index:6])):
+                self.board.change_board_level_2(index)
+                return True
+            if index > step and p2.get_sign() in self.board.get_board()[index]:
+                self.board.change_board(index, -(step + 1), player_now.get_sign())
+
+        elif ((abs(index - 23) == step and p1.get_sign() in self.board.get_board()[index]) or (
+                (abs(index - 23) + 1) < step and p1.get_sign() not in self.board.get_board()[18:index])):
+            self.board.change_board_level_2(index)
+            return True
+        if abs(index - 23) > step and p1.get_sign() in self.board.get_board()[index]:
+            self.board.change_board(index, (step + 1), p1.get_sign())
+            return True
+
+        return False
+
+
+def change_dict_val_cubes(dict_cubes, step):
+    if len(dict_cubes.keys()) > 1:
+        try:
+            dict_cubes[step]
+            dict_cubes[step] = 0
+        except KeyError:
+            for i in list(dict_cubes.keys()):
+                dict_cubes[i] = 0
+    else:
+        val = list(dict_cubes.keys())[0]
+        dict_cubes[val] = dict_cubes[val] - int(step / val)
+    return dict_cubes
+
+
+def dict_val_cubes(val_cubes):
+    if val_cubes[0] != val_cubes[1]:
+        dict_cubes = {val_cubes[0]: 1, val_cubes[1]: 1}
+    else:
+        dict_cubes = {val_cubes[0]: 4}
+    return dict_cubes
+
+
+def check_steps_regular(dict_cubes, step):
+    if ((step in dict_cubes and dict_cubes[step] != 0) or (
+            step == sum(dict_cubes) and sum(list(dict_cubes.values())) == 2)):
+        return True
+    return False
+
+
+def check_steps_when_double(dict_cubes, step):
+    val = int(list(dict_cubes.keys())[0])
+    if step == val and dict_cubes[step] != 0:
+        return True
+    if step % val == 0:
+        n = int(step / val)
+        if n in range(1, dict_cubes[val] + 1):
+            return True
+    return False
+
+
+def check_steps(dic_val_cubes, step):
+    if len(dic_val_cubes) > 1:
+        return check_steps_regular(dic_val_cubes, step)
+    return check_steps_when_double(dic_val_cubes, step)
+
+
+def print_str_board(board):
+    asterisk_arr = ['*' for i in range(26)]
+    print(" ", end=" ")
+    print(*asterisk_arr[:13])
+    print_first_half_board(board[:12])
+    print_second_half_board(board[12:])
+    print(" ", end=" ")
+    print(*asterisk_arr[13:])
+
+
+def get_len(arr):
```

Comment:
we don't use array, we use a list. but the argument name shouldn't be any type name